portfolio/src/main/java/com/google/sps/servlets/AuthStatusServlet.java

```diff
@@ -0,0 +1,48 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package com.google.sps.servlets;
+
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+/**
+ * Servlet to retrieve the authentication status of a user.
+ */
+@WebServlet("/auth-status")
+public class AuthStatusServlet extends HttpServlet {
+
+  // UserService handles User Authentication and Authentication Status
+  private final UserService userService = UserServiceFactory.getUserService();
+
+  /**
+   * Will return the authentication status of the current user.
+   * @param request HTTP request received from the client. Should contain no params.
+   * @param response HTTP Response to send back to the user. Either contains "true" or "false"
+   * @throws IOException if there is an issue with getWriter() while processing the request.
+   */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    boolean isLoggedIn = userService.isUserLoggedIn();
+
+    response.setContentType("application/json");
+    response.getWriter().println(isLoggedIn);
```

Comment:
Food for thought: technically a single boolean is a valid JSON document. However, it's almost always desirable to wrap your data in either an object or a list of objects (depending on if the URL points to a single entity or a collection). This comes in really useful if you later need to add more data to the authentication status. If you add new fields to an object, older clients will just ignore them. If you change the root element from a boolean to an object, it'll break older clients. 

(Not really an issue if your client is your JS code, but it would matter if you had a mobile client, or if others were using your API.)